storage/invoke_test.go

```diff
@@ -298,6 +298,68 @@ func (f *fakeApiaryRequest) Header() http.Header {
 	return f.header
 }
 
+func TestInvokeHeaderMerge(t *testing.T) {
```

Comment:
Could you add a comment explaining what this is testing for? Probably a little unintuitive to anyone who is not familiar.